flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/PrioritizedOperatorSubtaskState.java

```diff
@@ -281,10 +281,15 @@ public PrioritizedOperatorSubtaskState build() {
 				// approve-function signaled true.
 				if (alternative != null
 					&& alternative.hasState()
-					&& alternative.size() == 1
-					&& approveFun.apply(reference, alternative.iterator().next())) {
```

Comment:
Could we wrap this application in `BooleanUtils.isTrue(approveFun.apply(reference, alternative.iterator().next()))` in order to avoid the extra conditions?